### PR TITLE
Disable dynamic mapping in testSimpleGetFieldMappingsWithDefaults

### LINKED ISSUE #37928: CI: SimpleGetFieldMappingsIT testSimpleGetFieldMappingsWithDefaults fails with NPE
## Example build failure
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+corretto-periodic/ES_BUILD_JAVA=java11,label=amazon/79/console

## Reproduction line 
does not reproduce locally
```
./gradlew :server:integTest -Dtests.seed=797A064C3876272 -Dtests.class=org.elasticsearch.indices.mapping.SimpleGetFieldMappingsIT -Dtests.method="testSimpleGetFieldMappingsWithDefaults" -Dtests.security.manager=true -Dtests.locale=saq-KE -Dtests.timezone=Asia/Rangoon -Dcompiler.java=11 -Druntime.java=11
```

## Example relevant log:
```
ERROR   0.74s J6 | SimpleGetFieldMappingsIT.testSimpleGetFieldMappingsWithDefaults <<< FAILURES!
   > Throwable #1: java.lang.NullPointerException
   >    at __randomizedtesting.SeedInfo.seed([797A064C3876272:268BF528655F4903]:0)
   >    at org.elasticsearch.indices.mapping.SimpleGetFieldMappingsIT.testSimpleGetFieldMappingsWithDefaults(SimpleGetFieldMappingsIT.java:156)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   >    at java.base/java.lang.Thread.run(Thread.java:834)

```

## Frequency
Just this once, but I was able to reproduce it on master when running tests in a loop on a VM running centos 7 CI image. I'm still running it locally but no reproduction so far.

### PR DESCRIPTION
**Pull request by dnhatn**
Since #31140 we no longer require acking on the dynamic mapping of index requests. Thus, a returned mapping from a get mapping request does not necessarily contain the dynamic updates from the index request. This commit replaces the dynamic mapping update with a manual put mapping.

Relates #31140
Closes #37928

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/indices/mapping/SimpleGetFieldMappingsIT.java [hunk 1]
```diff
@@ -150,19 +150,27 @@ public void testSimpleGetFieldMappingsWithDefaults() throws Exception {
...
+            assertNotNull(response.fieldMappings("test", "type", "field1"));
+            assertThat((Map<String, Object>) response.fieldMappings("test", "type", "field1").sourceAsMap().get("field1"),
+                hasEntry("index", Boolean.TRUE));
+            assertThat((Map<String, Object>) response.fieldMappings("test", "type", "field1").sourceAsMap().get("field1"),
+                hasEntry("type", "text"));
+
+            assertNotNull(response.fieldMappings("test", "type", "obj.subfield"));
+            assertThat((Map<String, Object>) response.fieldMappings("test", "type", "obj.subfield").sourceAsMap().get("subfield"),
+                hasEntry("type", "keyword"));
+        });
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/indices/mapping/SimpleGetFieldMappingsIT.java]: I was wondering if the whole code block needs to be in the assertBusy ?

Comment (dnhatn) [reply server/src/test/java/org/elasticsearch/indices/mapping/SimpleGetFieldMappingsIT.java]: No, we only need to ensure that the mapping is ready. Anyway, I replaced the dynamic mapping update with a put-mapping request. Thanks for looking @javanna.

**Comment by dnhatn** [comment]
run elasticsearch-ci/default-distro

**Review by jtibshirani — APPROVED** [review]
This looks good to me!

**Comment by dnhatn** [comment]
Thanks @javanna and @jtibshirani.

**Pull request merged**